syntax = "proto3";
option java_multiple_files = true;
package eu.ifak.ehs;


message PingRequest {
}
message PingResponse {
    bool value = 1;
    Status status = 2;
}


message GetConfigurationRequest {
}
message GetConfigurationResponse {
    string value = 1;
    Status status = 2;
}


enum Status {
    SUCCESS = 0;
    ADAPTER_NOT_RESPONDING = 1;
    DEVICE_NOT_RESPONDING = 2;
    TYPE_MISMATCH = 3;
    UNKNOWN = 4;
}
message Value {
    oneof value {
        bool bool_value = 1;
        int64 int64_value = 2;
        double double_value = 3;
        string string_value = 4;
    }
}
message ValueAndStatus {
    Value value = 1;
    Status status = 2;
}
